qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/RoadObjectsActivity.kt

```diff
@@ -163,12 +161,16 @@ class RoadObjectsActivity : AppCompatActivity() {
             enabled = true
         }
         mapboxNavigation.registerRoutesObserver(routesObserver)
-        mapboxNavigation.setRoutes(listOf(getRoute()))
         mapboxNavigation.registerLocationObserver(locationObserver)
-        mapboxNavigation.registerRouteProgressObserver(replayProgressObserver)
-        mapboxReplayer.pushRealLocation(this, 0.0)
-        mapboxReplayer.playbackSpeed(1.5)
-        mapboxReplayer.play()
+    }
+
+    @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+    private fun initReplayer() {
+        mapboxTripStarter.onAttached(mapboxNavigation)
+        val routeOrigin = Utils.getRouteOriginPoint(getRoute())
+        mapboxNavigation.mapboxReplayer.pushEvents(
+            listOf(ReplayRouteMapper.mapToUpdateLocation(0.0, routeOrigin))
+        )
```

Comment:
I'd like to move this into `ReplayRouteSession`. Whenever the route changes, we can emit the origin point into the replayer.

Reply:
https://github.com/mapbox/mapbox-navigation-android/pull/6817#discussion_r1066876568

Similar comment from the other pull request. If we push the origin point and reset the navigator, we could potentially change routes and move to the origin.

I can change that as part of this pull request tomorrow

Reply:
I'm considering this out of scope of this pull request now. I just worked on it and it created a few issues, and there is a bit of api and logic to define it in the replay session. Will propose it as a follow up